src/tests/bptree.test.ts

```diff
@@ -1,49 +1,127 @@
-import { BPTree, MetaPage } from "../btree/bptree";
+import {
+	BPTree,
+	MetaPage,
+	ReferencedValue,
+	binarySearchReferencedValues,
+} from "../btree/bptree";
 import { MemoryPointer } from "../btree/node";
+import { IndexFileV1 } from "../index-file/index-file";
 import { RangeResolver } from "../resolver";
+import { readBinaryFile } from "./test-util";
 
-function textEncode(phr: string): Uint8Array {
-  const encoder = new TextEncoder();
+class testMetaPage implements MetaPage {
+	private rootMP: MemoryPointer;
 
-  return encoder.encode(phr);
+	constructor(mp: MemoryPointer) {
+		this.rootMP = mp;
+	}
+
+	async root(): Promise<MemoryPointer | null> {
+		return this.rootMP;
+	}
 }
 
-class TestMetaPage implements MetaPage {
-  private _root: MemoryPointer;
+describe("test compare bytes", () => {
+	let mockRangeResolver: RangeResolver;
+	let indexFileSize: number;
+	let mockDataFileResolver: RangeResolver;
 
-  constructor(initialRoot: MemoryPointer) {
-    this._root = initialRoot;
-  }
+	beforeEach(() => {
+		mockRangeResolver = async ({ start, end }) => {
+			const indexFile = await readBinaryFile("bptree_1.bin");
 
-  setRoot(mp: MemoryPointer): void {
-    this._root = mp;
-  }
+			const slicedPart = indexFile.slice(start, end + 1);
 
-  root(): MemoryPointer {
-    return this._root;
-  }
-}
-
-describe("test compare bytes", () => {
-  let buffer: Uint8Array;
-  let resolver: RangeResolver;
+			const arrayBuffer = slicedPart.buffer.slice(
+				slicedPart.byteOffset,
+				slicedPart.byteOffset + slicedPart.byteLength
+			);
 
-  beforeEach(() => {
-    buffer = new Uint8Array(4096);
-  });
+			return {
+				data: arrayBuffer,
+				totalLength: arrayBuffer.byteLength,
+			};
+		};
 
-  it("generates an empty tree", async () => {
-    /*
-		let mp: MemoryPointer = {
-			offset: BigInt(0),
-			length: 0,
+		mockDataFileResolver = async ({ start, end }) => {
+			return {
+				data: new ArrayBuffer(0),
+				totalLength: 0,
+			};
 		};
+	});
+
+	it("should read a bptree", async () => {
+		const page = new testMetaPage({ offset: 0n, length: 1 });
+		const bptree = new BPTree(mockRangeResolver, page, mockDataFileResolver);
+
+		const textEncoder = new TextEncoder();
+		const helloBuffer = textEncoder.encode("hello");
+		const key = new ReferencedValue(
+			{ offset: 0n, length: 1 },
+			helloBuffer.buffer
+		);
+
+		// const [rv, mp] = await bptree.find(key);
 
-		const tree = new BPTree(resolver, new TestMetaPage(mp), buffer);
+		// console.log(rv, mp);
+	});
 
-		const res = await tree.find(textEncode("howdy"));
+	it("should compareBytes correctly", () => {
+		const buffer1 = new Uint8Array([0]).buffer;
+		const buffer2 = new Uint8Array([0]).buffer;
 
-		expect(res).toEqual([{ offset: BigInt(0), length: 0 }, false]);
-		*/
+		const result = ReferencedValue.compareBytes(buffer1, buffer2);
+		console.log(result); 
+
+    expect(result).toEqual(0)
   });
+
+	it("should compare reference values", () => {
```

Comment:
I've asserted that this function behaves accordingly by writing out the exact test coverage for the go implementation.

`Found` will only return `true` iff the `value` exists and the `MemoryPointer` is that exact one.